content/slice/language-guide/enum-types.md

```diff
@@ -135,48 +212,92 @@ public enum Fruit : int
 }
 ```
 {% /side-by-side %}
+
+The underlying type of the mapped enumeration is always `int`.
 {% /slice1 %}
 
 {% slice2 %}
+### Enum with underlying type in C#
+
+An enum with underlying type maps to a public C# enumeration with the same name, and each Slice enumerator maps to the
+C# enumerator with the same name. For example:
+
 {% side-by-side alignment="top" %}
+
 ```slice
 enum Fruit : uint8 { Apple, Pear, Orange }
 ```
 
 ```csharp
-public enum Fruit : byte
+public enum Fruit : uint8
 {
     Apple = 0,
     Pear = 1,
-    Orange = 2,
+    Orange = 2
 }
 ```
 {% /side-by-side %}
-{% /slice2 %}
-
-{% slice1 %}
-The underlying type of the mapped enumeration is always `int`.
-{% /slice1 %}
 
-{% slice2 %}
 The underlying type of the mapped enumeration is always the mapped type for the Slice underlying type. For example,
 a Slice `uint8` corresponds to a C# `byte`.
-{% /slice2 %}
 
-The Slice compiler also generates 3 extension classes for this enumeration, each with an extension method:
- - As*Name* to convert an integral value into an enum instance (with enumerator-checking for checked enums)
- - Encode*Name* to encode an enum instance
- - Decode*Name* to decode an enum instance
+### Enum with fields in C#
 
-With our `Fruit` example, we get:
+C# does not provide a native discriminated union type. This mapping relies on an emulation based on record classes
+[proposed a few years ago]. Briefly, a Slice enum with fields maps to a C# partial record class with the same name, and
+each enumerator maps to a public nested record class with the same name as the enumerator. This nested record class
+derives from the enclosing "enum type" record class.
+
+The Slice code generator adds the attribute `[Dunet.Union]` to the base record class, which instructs the [Dunet] source
+generator to generate additional helper code for these partial records.
+
+For example:
+
+{% side-by-side alignment="top" %}
+
+```slice
+enum Shape {
+    Circle(radius: uint32)
+    Rectangle(width: uint32, length: uint32)
+    Dot
+}
+```
 
-{% slice1 %}
 ```csharp
-public static class FruitByteExtensions
+[Dunet.Union]
+public partial record class Shape
```

Comment:
Should we also include the fact that this is `abstract`?

Reply:
I guess we can, to emphasize you can't create a plain Shape.

Keep in mind that Dunet adds abstract sealed etc. as appropriate.

Reply:
> to emphasize you can't create a plain Shape.

Yes, that was my thinking. Otherwise what would it mean to create a bare `Shape`?

> Keep in mind that Dunet adds abstract sealed

For our Slice enums though, they'll always be abstract, correct? Or is there some way it would map to a concrete class?

Reply:
Added abstract.